### ISSUE TITLE
Can not access Admiral anymore after running for a while in OVA

### ISSUE THREAD

**Issue by ywk253100** [vmware/vic-product#483]
I deployed VIC from OVA and can not access Admiral after running for a while(about 1 or 2 hours). When trying to access https://ip:8282, the browser was redirected to the URL as following and got a blank page: 
https://vc-ip/websso/SAML2/SSO/vsphere.local?SAMLRequest=zVTfb9owEP5XIr8ndkwpYAFVV1atUrtmhU3TXibHuYKlxM58JqH%2F%2FZwAK%2Bqmisc9%2BMW%2B%2B%2B77cfL0aleVUQMOtTUzkiaMRGCULbRZz8jX1W08JlfzKcqqrMX11m%2FME%2FzaAvoo9BkU%2FcOMbJ0RVqJGYWQFKLwSy%2BuHe8ETJmpnvVW2JNEi9GkjfT9q432NglJUPHaFrTFuKnYZFxtVxzydxJwlYNZJU7XSQaJsRVvIES3tgDldLh9pg%2FUGwmNplQzwt9Yp6DnOyLMsEUh0t5iRn5d8OCykzGGicsVHIxhdsiK%2FmIz46GIs8zyUYSYRdQOvjYhbuDPopfEzwlk6itk4ZumKTcSAiyFLBin%2FQaLsIO6DNnvL3nMi3xeh%2BLRaZXH2uFz1AI0uwH0O1cF%2FlqQpT1IezlCM%2BZiT6NsxnABBjlH0%2FNz5IQR54Drjyfxo%2FN%2BzaAd0tLkCLwvp5ZSeTtzP57Xo%2BN4tMltq9XJCg5%2B9DNdladsbB9IH3d5toQ%2Bwkv59gO5GF%2FFzXyrqzhv0YDyJllnH6ctWlvpZg3vdsDOFEnoQJ25syKkzC0%2BVnW3wW5QDSBNa9qQCp5O1RrWBSiKV3ru4B6Zh3zhlF%2FTjLkjrwkdyANmh%2FoPRtm3SDhLr1qGBpfT7w%2F2yx4p1v7gqWBrqhX%2Bpg8XdePEEBlqZl7AKd%2F8Q%2FB9RXUAJ61Oq9G048%2BNqnv5L898%3D&RelayState=SessionId&SigAlg=http%3A%2F%2Fwww.w3.org%2F2001%2F04%2Fxmldsig-more%23rsa-sha256&Signature=u1uHQCoaGiXtKMQcg8AxIBwuKxof8tvq1zyb3qf8G%2FBAdehHJ5sLa5oPvqAKEEMxuJypQYBUT90ppxrYb3GGnPJbOaPPt0BQu2AHDBx08Ensqa8CXtY3zhDvbz66ZKVBW%2FZUcoQT1KCfgKLyiiB9%2BlAfGc5zsspklagIhXD8vWHFBIFefKJdGT1lvVGT%2BZ6ULk0sNMWEuDSle9O4mM5V5nqXCWp25ILLna68IL89dv5%2FBurgyIVJaiRM0z5OTOKM600KAC2jW%2FGbCEaanau0JdejL1gseDcOTVjUk20Ue%2F7GfymV75z8kV8lNGbMDuA2csjKtFK5W%2BVk%2ByvScDcjmQ%3D%3D

The logs of Admiral:
```
[ForkJoinPool-1-worker-0] INFO com.vmware.admiral.auth.idm.psc.saml.sso.authentication.SamlRequestSender - SP alias for the login request is 10.112.122.15:8282
[ForkJoinPool-1-worker-0] INFO com.vmware.identity.websso.client.endpoint.SsoRequestSender - Producing redirect url
[ForkJoinPool-1-worker-0] WARN com.vmware.identity.websso.client.SiteAffinity - Failed to init CdcSession. likely due to missing vmafd jar. Message: com/vmware/identity/cdc/CdcFactory
[ForkJoinPool-1-worker-0] INFO com.vmware.identity.websso.client.endpoint.SsoRequestSender - Added Renewable condition
[ForkJoinPool-1-worker-0] INFO com.vmware.identity.websso.client.endpoint.SsoRequestSender - Added Delegable condition
[ForkJoinPool-1-worker-0] INFO com.vmware.identity.websso.client.endpoint.SsoRequestSender - Destination URL: https://sc2-rdops-vm06-dhcp-219-20.eng.vmware.com/websso/SAML2/SSO/vsphere.local
[ForkJoinPool-1-worker-0] INFO com.vmware.identity.websso.client.endpoint.SsoRequestSender - Relay State value is: SessionId
```

**Comment by anchal-agrawal** [comment]
Hi @ywk253100, those logs don't show any fatal errors. Can you post a larger log snippet?

In a functioning scenario, if the Admiral page has been left open for a while, the user sees s message at the top of the UI asking to reload the page and log in again.

**Comment by steven-zou** [comment]
Logs from Admiral container:
======
[316][I][2017-08-04T05:21:48.000Z][11][8282/resources/epz-periodic-enumeration][doTrigger][Periodic refresh]
[https://172.17.0.2:8282/ForkJoinPool-1-worker-3] INFO com.vmware.admiral.auth.idm.psc.saml.sso.authentication.SamlRequestSender - SP alias for the login request is 10.112.122.54:8282
[https://172.17.0.2:8282/ForkJoinPool-1-worker-3] INFO com.vmware.identity.websso.client.endpoint.SsoRequestSender - Producing redirect url
[https://172.17.0.2:8282/ForkJoinPool-1-worker-3] WARN com.vmware.identity.websso.client.SiteAffinity - Failed to init CdcSession. likely due to missing vmafd jar. Message: com/vmware/identity/cdc/CdcFactory
[https://172.17.0.2:8282/ForkJoinPool-1-worker-3] INFO com.vmware.identity.websso.client.endpoint.SsoRequestSender - Added Renewable condition
[https://172.17.0.2:8282/ForkJoinPool-1-worker-3] INFO com.vmware.identity.websso.client.endpoint.SsoRequestSender - Added Delegable condition
[https://172.17.0.2:8282/ForkJoinPool-1-worker-3] INFO com.vmware.identity.websso.client.endpoint.SsoRequestSender - Destination URL: https://10.112.123.52/websso/SAML2/SSO/vsphere.local
[https://172.17.0.2:8282/ForkJoinPool-1-worker-3] INFO com.vmware.identity.websso.client.endpoint.SsoRequestSender - Relay State value is: SessionId

**Comment by reasonerjt** [comment]
@anchal-agrawal 
Maybe we need some guys from PSC to help us debug, look at the URL it seemed to be blocked on the IP of vcenter?

**Comment by reasonerjt** [comment]
I reproduced this issue on a recent build: **vic-b5fc5989-dev.ova**
Browser was redirected to: https://sc2-rdops-vm03-dhcp-106-152.eng.vmware.com/websso/SAML2/SSO/vsphere.local?SAMLRequest=zVRbb5swFP4ryO9gcEIuVpIqa1atUruyhk3TXiYHThJLYDMfB9J%2FP0OSFXVTlce94nO%2B893E7OZYFl4NBqVWcxIFIfFAZTqXajcnX9M7f0JuFjMUZVHx5cHu1TP8OgBaz%2B0p5N3DnByM4lqgRK5ECchtxtfLxwfOgpBXRlud6YJ4K7cnlbDdqb21FXJKMWO%2ByXWFfl2GAz%2FfZ5UfhSM%2FilkAahfUZSMMBJkuaQMbRE1bZEbX6ydaY7UH91joTDj8O20y6EjOyVYUCMS7X83JzwGLIWcTMYxG0w0bbYdxPB0Np2IUQTQQ4caNYSIQZQ2vi4gHuFdohbJzwsJo7IcTP4rScMrjmMejIB6PfxAvOav7INXJs%2Fes2JyGkH9K08RPntZpB1DLHMxnN%2B0CCINoyoLhIIgGEz5hE0a8b5d0HAS5ZNHxM9en4OSBaZ0ni4vzf9%2BiLdDF5hKsyIUVM9q%2FeLrPKt7yvV8lupDZS48Gu7oNy6LQza0BYZ1uaw7QBVgK%2Bz5A%2B0Xm%2FrYb5VXrDVpQlnjrpOX05SAKuZVgXit2pVBCz%2BL4rXY5tWZhX9nVBr9FOYPUbuVEynHq1RqzPZQCqbDW%2BB0wdX1jNBzSj0cnrQ0fyRnkiPIPRtM0QTMItNm5hTCi3x8f1h2WL7viZs5SN8%2FtS%2BUsbs%2FzZ1DQiE0Bqfv2D8H%2FEdUVFLDrU6Vvw1lcqtn%2FMS1%2BAw%3D%3D&RelayState=SessionId&SigAlg=http%3A%2F%2Fwww.w3.org%2F2001%2F04%2Fxmldsig-more%23rsa-sha256&Signature=oBFPJ5GuNDZ9Zb5UOu8ZDtUppU1Yr4cca4Zbljsk3xP%2B%2Fm9rgV3e%2F%2FsQrv1sHOSv00V4xorKDb%2B%2B1GbQpPPONx30fK%2BFXZ0Im1FKOpG6qfaGAcVfGoZ56rFNfmks5DZ9ovC2TjvvQLDVnjDFTL3mm%2B%2B4uEi6CGGhM3hWOjrULOMcDMlwWqLfFAG8jwhcyIC7p3gM5elZQ6YwprCFzDvvw8Eo634bau0HOnS%2FYKjL6R%2FZuN5Uwmtq8c%2Fz1c3%2B%2Bub0xZ%2FQHxL67JyK4qHFGbofkC4oOdbbclRbUHDbJtJmQ4O05qizd%2FVn2KH5OgHBOvuXwa1zJ0%2BPva3zx%2BFEXTAwlw%3D%3D

And showed a blank page.

I tried to curl this url, the response was "200" but has a header "CastleError: SW52YWxpZCBSZXF1ZXN0IFNpZ25hdHVyZQ=="

Decoded the value via base64 it seems to be "Invalid Request Signature"

@anchal-agrawal we may need to investigate if this due to bug in Admiral or misconfiguration?

**Comment by lcastellano** [comment]
This URL should bring up the PSC "Sign ON" page. The URL is generated within Admiral by calling the PSC identity-client library. That library needs certs from the PSC, if the PSC updates the certs the URL may be incorrect. How often have you seen this failure?

**Comment by mdubya66** [comment]
making this a high priority. needs detailed triage.

**Comment by reasonerjt** [comment]
@lcastellano 
We see this issue repeatedly on one VM, if I reboot this VM the problem is gone, keep it running for a couple of hours, the problem re-occurs.

I didn't have chance to deploy another VM with the same build.

**Comment by stuclem** [comment]
@alextopuzov and I are seeing the same issue with our setup, and @andrewtchin asked that I report our experience here.

 After deploying `chin-vic-c2506807-dev-upgrade.ova` this morning, I was able to use SSO credentials to successfully log in to the Management Portal at port 8282. After a certain time (cannot say for sure how long, maybe 45 minutes, maybe less) the UI seemed to become unresponsive while I was importing users. Then, after attempting to reload the page, close the browser tab, then log back in in a new tab, the UI gets stuck at the SSO redirect, and I just see a white page. I have not been able to access the management portal since. @alextopuzov has not been able to log in at all, and has only ever seen the frozen SSO redirect. 

I never managed to log in at all on Friday with the previous build, `chin-vic-70e444e8-dev.ova`.

Restarting the VC VM and the VIC appliance, clearing browser cache, using different browsers, all have the same result.

If this is related to the PSC, we are using vCenter Server 6.5u1 running on Windows Server 2008, with an embedded PSC.

**Comment by morris-jason** [comment]
Saw this again last week in my test environment. Same behavior that @stuclem reported.

**Comment by andrewtchin** [comment]
@sergiosagu will be triaging this issue tomorrow

**Comment by stuclem** [comment]
Trying this again with the newest build (`vic-0963d9b5-dev.ova`). It seems that the first ever login to the management portal always works. Subsequent logins result in the white screen described above.

**Comment by stuclem** [comment]
I deployed the latest OVA build today. The bug wherein you had to log in to perform the initialization of the OVA the first time you visited the Getting Started page seems to have been fixed (thanks @andrewtchin !). 

It appears that this issue was the cause of the blank screen on later attempts to log in to the management portal, as @sergiosagu suspected. @alextopuzov and I both just successfully logged into the same management portal instance with the latest build of the OVA.
